lib/promisepay_ex/client.ex

```diff
@@ -59,4 +66,42 @@ defmodule PromisepayEx.Client do
   defp build_params(params, url) do
     to_char_list(url <> "?" <> params)
   end
+
+  def perform_post(_url, _params, nil, nil) do
+    raise(PromisepayEx.Error, message: "Error")
+  end
+
+  @spec perform_post(String.t, Keyword.t, String.t, String.t) :: Map.t
+  def perform_post(url, params, username, password) do
+    params
+    |> URI.encode_query
+    |> build_params(url)
+    |> send_post_request(username, password)
+  end
+
+  defp send_post_request(request, username, password) do
+    headers = build_authorization_header(username, password)
+    response = HTTPoison.post(request, "", headers)
+
+    case response do
+      {
+        :ok,
+        %HTTPoison.Response{
+          status_code: 200,
+          body: body,
+          headers: headers
+        }
+      } ->
+        {:ok, {200, headers, body}}
+      {:ok, %HTTPoison.Response{status_code: 422}} ->
+        {:error, "Not Found"}
+      {:error, %HTTPoison.Error{reason: reason}} ->
+        {:error, reason}
+    end
+  end
+
+  defp build_authorization_header(username, password) do
```

Comment:
maybe we could create a config option so that the user can choose between basic authorization and header authorization? Not the scope of this PR though